src/databricks/labs/ucx/azure/azure_credentials.py

```diff
@@ -0,0 +1,283 @@
+import base64
+import logging
+from collections import namedtuple
+from dataclasses import dataclass
+
+from databricks.labs.blueprint.installation import Installation
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import InternalError, ResourceDoesNotExist
+from databricks.sdk.errors.platform import InvalidParameterValue
+from databricks.sdk.service.catalog import (
+    AzureServicePrincipal,
+    Privilege,
+    StorageCredentialInfo,
+    ValidateStorageCredentialResponse,
+    ValidationResult,
+)
+
+from databricks.labs.ucx.assessment.azure import AzureServicePrincipalCrawler
+from databricks.labs.ucx.azure.access import (
+    AzureResourcePermissions,
+    StoragePermissionMapping,
+)
+from databricks.labs.ucx.azure.resources import AzureResources
+from databricks.labs.ucx.config import WorkspaceConfig
+from databricks.labs.ucx.framework.crawlers import StatementExecutionBackend
+from databricks.labs.ucx.hive_metastore.locations import ExternalLocations
+
+logger = logging.getLogger(__name__)
+
+
+# A namedtuple to host service_principal and its client_secret info
+ServicePrincipalMigrationInfo = namedtuple("ServicePrincipalMigrationInfo", "service_principal client_secret")
+
+
+@dataclass
+class StorageCredentialValidationResult:
+    name: str
+    azure_service_principal: AzureServicePrincipal
+    created_by: str
+    read_only: bool
+    results: list[ValidationResult]
+
+    @classmethod
+    def from_storage_credential_validation(
+        cls, storage_credential: StorageCredentialInfo, validation: ValidateStorageCredentialResponse
+    ):
+        return cls(
+            name=storage_credential.name or "",
+            azure_service_principal=storage_credential.azure_service_principal or AzureServicePrincipal("", "", ""),
+            created_by=storage_credential.created_by or "",
+            read_only=storage_credential.read_only or False,
+            results=validation.results or [],
+        )
+
+
+class AzureServicePrincipalMigration:
```

Comment:
Can you remove Azure prefix from classes defined in this file, as this is clear it's about Azure 😉

Reply:
fixed